test/services/sync/LdapSync.integration.test.js

```diff
@@ -0,0 +1,310 @@
+	this.timeout(20000);
+	let app;
+	let server;
+	let system;
+
+	const cleanupAll = async () => {
+		await testObjects.cleanup();
+		const accounts = exampleLdapUserData.map((user) => `${user.schoolDn}/${user.ldapUID}`.toLowerCase());
+		await accountModel
+			.deleteMany({ username: { $in: accounts } })
+			.lean()
+			.exec();
+		const userLdapDns = exampleLdapUserData.map((user) => user.ldapDn);
+		await userModel
+			.deleteMany({ ldapDn: { $in: userLdapDns } })
+			.lean()
+			.exec();
+		const classLdapDns = exampleLdapClassData.map((clazz) => clazz.ldapDN);
+		await classModel
+			.deleteMany({ ldapDN: { $in: classLdapDns } })
+			.lean()
+			.exec();
+		const schoolLdapDns = exampleLdapSchoolData.map((school) => school.ldapOu);
+		await schoolModel
+			.deleteMany({ ldapSchoolIdentifier: { $in: schoolLdapDns } })
+			.lean()
+			.exec();
+	};
+
+	before(async () => {
+		app = await appPromise;
+		server = await app.listen(0);
+		await cleanupAll();
+	});
+
+	beforeEach(async () => {
+		system = await testObjects.createTestSystem({
+			type: 'ldap',
+			ldapConfig: fakeLdapConfig,
+			alias: SYSTEM_ALIAS,
+		});
+	});
+
+	afterEach(async () => {
+		await cleanupAll();
+	});
+
+	after(async () => {
+		await server.close();
+	});
+
+	async function executeLdapSync(testFuncContext, options) {
+		const params = { query: { target: 'ldap' } };
+		app.services.ldap = new FakeLdapService(app, options);
+		try {
+			const result = await app.service('sync').find(params);
+			if (!isRabbitMqAvailable(result)) {
+				testFuncContext.skip();
+			}
+			// eslint-disable-next-line promise/param-names
+			await new Promise((r) => setTimeout(r, 1000));
+			expect(result.length).to.be.eql(1);
+			expect(result[0].success).to.be.true;
+		} catch (e) {
+			syncLogger.error(e);
+		}
+	}
+
+	it('should create schools from ldap sync', async function test() {
+		const options = {
+			fillSchools: true,
+		};
+		await executeLdapSync(this, options);
+
+		const foundSchools = await Promise.all(
+			exampleLdapSchoolData.map((exampleSchool) =>
+				SchoolRepo.findSchoolByLdapIdAndSystem(exampleSchool.ldapOu, system._id)
+			)
+		);
+		expect(foundSchools.length).to.be.eql(exampleLdapSchoolData.length);
+		foundSchools.forEach((foundSchool) => expect(foundSchool).to.be.not.null);
+	});
+
+	it('should create users from ldap sync', async function test() {
+		const schools = await createTestSchools(system);
+
+		const options = {
+			fillSchools: true,
+			fillUsers: true,
+		};
+		await executeLdapSync(this, options);
+
+		const foundUsers = await Promise.all(
+			exampleLdapUserData.map((exampleUser) => {
+				const foundSchool = schools.filter((school) => school.ldapSchoolIdentifier === exampleUser.schoolDn);
+				return UserRepo.findByLdapIdAndSchool(exampleUser.ldapUUID, foundSchool[0]._id);
+			})
+		);
+		expect(foundUsers.length).to.be.eql(exampleLdapUserData.length);
+		foundUsers.forEach((foundUser) => expect(foundUser).to.be.not.null);
```

Comment:
https://www.chaijs.com/plugins/chai-each/  is this is a option?

Reply:
That looks very promising, but it is not used in our project yet. We will look at it later.

Reply:
The problem is that the forEach not work well in some situations. Because mocha build first the test tree then run the test. The tree maybe not include the expects but the second point is that it do not throw errors well to the results. (in older mocha versions, maybe it is fixed in the current).